fuse/fuse.cpp

```diff
@@ -101,8 +155,16 @@ int run_in_fuse(
 	// We can safely release the global handle now that we hold a livefd
 	(void)close(ffd);
 
+	// The fuse-waked process takes an input file containing visible files, json formatted.
 	std::ofstream ijson(ipath);
-	ijson.write(json.data(), json.size());
+	bool first = true;
+	ijson << "{ \"visible\": [";
+	for (auto &s : args.visible) {
+		ijson << (first ? "\"" : ", \"") ;
+		ijson << s << "\"";
```

Comment:
You did not escape `s`... Call either json_escape, or use the JSON construction APIs ala bsp/bsp.cpp